src/test/java/domain/CarsTest.java

```diff
@@ -0,0 +1,71 @@
+package domain;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import utils.NumberGenerator;
+
+import java.util.List;
+
+import static domain.Car.MINIMUM_NUMBER_TO_MOVE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class CarsTest {
+
+    private final NumberGenerator mockNumberGenerator = new MockNumberGenerator();
+
+    @DisplayName("moveCars를 통해 조건을 충족시킨 Car를 move 시킨다.")
+    @ParameterizedTest
+    @CsvSource(value = {"0:1", "1:0", "2:1"}, delimiter = ':')
+    void test1(int index, int expectedPosition) {
+        Car carA = new Car("carA");
+        Car carB = new Car("carB");
+        Car carC = new Car("carC");
+
+        Cars cars = new Cars(List.of(carA, carB, carC));
+
+        cars.moveCars(mockNumberGenerator);
+
+        Car carAfterMove = cars.getCars().get(index);
+        assertEquals(carAfterMove.getPosition(), expectedPosition);
+    }
+
+    @DisplayName("findWinners()를 통해 position이 가장 높은 Car들을 가져온다")
+    @Test
+    void test2() {
+        Car carA = new Car("carA");
+        Car carB = new Car("carB");
+        Car carC = new Car("carC");
+
+        Cars cars = new Cars(List.of(carA, carB, carC));
+
+        cars.moveCars(mockNumberGenerator);
+        List<Car> winners = cars.findWinners();
+
+        assertThat(winners).containsExactly(carA, carC);
+    }
+
+    /**
+     * 홀수번은 move가 가능한 숫자를, 짝수번에는 move가 불가능한 숫자를 반환하는 NumberGenerator
+     */
+    class MockNumberGenerator implements NumberGenerator {
+
+        private List<Integer> values = List.of(MINIMUM_NUMBER_TO_MOVE , MINIMUM_NUMBER_TO_MOVE-1);
+        private int index = 0;
+
+        @Override
+        public int generate() {
+            Integer value = values.get(index);
+
+            if (index == 0) {
+                index++;
+                return value;
+            }
+
+            index--;
+            return value;
+        }
+    }
```

Comment:
NumberGenerator라는 인터페이스를 활용함으로써 랜덤값에 대한 의존을 덜어낸 것은 아주 좋습니다! 덕분에 테스트 시에 랜덤을 신경쓰지 않아도 되겠네요 👍

하지만 테스트용 구현체 코드가 너무 복잡하게 구성되어 있습니다. 해당 코드를 사용하는 사람은 `홀수번`엔 move, `짝수번`엔 non move라는 추가적인 규칙을 알아야 합니다. test1 코드는 원래 검증하려고 했던 Cars의 규칙 뿐만 아니라, MockNumberGenerator의 구현을 함께 고려해야 하므로 이 규칙을 덜어주는 게 좋을 것 같아요. 
해당 클래스에서 인덱스를 변경해가며 generate하던 부분을 제거하고  `List<Integer> values` 를 생성자로 주입받으면 어떨까요?
```suggestion
    class MockNumberGenerator implements NumberGenerator {

        private final List<Integer> values;
        private final index = 0;
        
        public MockNumberGenerator(List<Integer> values){
            this.values = values;
        }
        
        @Override
        public int generate() {
           return values.get(index++);
        }
    }
```